dali/image/generic_image.cc

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdlib>
+
+#include "dali/image/generic_image.h"
+#include "dali/image/png.h"
+
+namespace dali {
+
+bool CheckIsGIF(const uint8_t *gif, int size) {
+    DALI_ASSERT(gif);
+    return (size >= 10 && gif[0] == 'G' && gif[1] == 'I' && gif[2] == 'F' && gif[3] == '8' &&
+    (gif[4] == '7' || gif[4] == '9') && gif[5] == 'a');
+}
+
+// OpenCV doesn't handle gif images so we don't need it now
+#if 0
+DALIError_t GetGIFImageDims(const uint8 *gif, int size, int *h, int *w) {
+    DALIError_t ret = DALIError;
+    DALI_ASSERT(gif);
+
+    if (size >= 10) {
+        *w = (unsigned int)(gif[6] | gif[7] << 4) & 0xFFFF;
+        *h = (unsigned int)(gif[8] | gif[9]<< 4) & 0xFFFF;
+        ret = DALISuccess;
```

Comment:
At this point might be worth checking that w*h is the same as size minus the length of the GIF header. Might help catching malformed images or incorrect allocations

Reply:
No really, as size parameter her is the size of the data we have read, and as GIF is compressed file, so w*h is suppose to be bigger than the file size.